src/agent/state-inspector.ts

```diff
@@ -0,0 +1,549 @@
+    // TODO: Handle the case where `this.config_.workingDirectory` is `null`.
+    return path.indexOf(this.config_.workingDirectory as string) === 0;
+  }
+
+  isPathInNodeModulesDirectory_(path: string): boolean {
+    return path.indexOf('node_modules') === 0;
+  }
+
+  resolveFrame_(frame: inspector.Debugger.CallFrame, underFrameCap: boolean):
+      apiTypes.StackFrame {
+    let args: Array<apiTypes.Variable> = [];
+    let locals: Array<any> = [];
+
+    if (!underFrameCap) {
+      args.push({
+        name: 'arguments_not_available',
+        varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+      });
+      locals.push({
+        name: 'locals_not_available',
+        varTableIndex: ARG_LOCAL_LIMIT_MESSAGE_INDEX
+      });
+    } else {
+      locals = this.resolveLocalsList_(frame);
+
+      if (isEmpty(locals)) {
+        locals = [];
+      }
+    }
+    return {
+      function: this.resolveFunctionName_(frame),
+      location: this.resolveLocation_(frame),
+      arguments: args,
+      locals: locals
+    };
+  }
+
+  resolveFunctionName_(frame: inspector.Debugger.CallFrame): string {
+    if (!frame) {
+      return '';
+    }
+    if (frame.functionName === '') {
+      return '(anonymous function)';
+    }
+    return frame.functionName;
+  }
+
+  resolveLocation_(frame: inspector.Debugger.CallFrame):
+      apiTypes.SourceLocation {
+    return {
+      path: this.resolveRelativePath_(frame),
+      line: frame.location.lineNumber
+    };
+  }
+
+  /**
+   * Iterates and returns variable information for all scopes (excluding global)
+   * in a given frame. FrameMirrors should return their scope object list with
+   * most deeply nested scope first so variables initially encountered will take
+   * precedence over subsequent instance with the same name - this is tracked in
+   * the usedNames map. The argument list given to this function may be
+   * manipulated if variables with a deeper scope occur which have the same
+   * name.
+   * @function resolveLocalsList_
+   * @memberof StateResolver
+   * @param {inspector.Debugger.CallFrame} frame - A instance of callframe.
+   * @returns {Array<Object>} - returns an array containing data about selected
+   *  variables
+   */
+  resolveLocalsList_(frame: inspector.Debugger.CallFrame): apiTypes.Variable[] {
+    let locals: Array<any> = [];
+
+    const usedNames: {[name: string]: boolean} = {};
+    const allScopes = frame.scopeChain;
+    let count = allScopes.length;
+    // We find the top-level (module global) variable pollute the local
+    // variables we omit them by default, unless the breakpoint itself is
+    // top-level. The last scope is always omitted.
+    if (frame.scopeChain[count - 2].type === 'closure')
+      count -= 2;
+    else
+      count -= 1;
+    for (let i = 0; i < count; ++i) {
+      let result = this.v8Inspector_.getProperties(
+          frame.scopeChain[i].object.objectId as string);
+      if (result.response && !isEmpty(result.response.result)) {
```

Comment:
Add `!result.error` here, and a TODO to handle this error if it exists. Right now it seems like it's getting silently ignored.

Reply:
Done.